packages/flutter/lib/src/material/date.dart

```diff
@@ -8,9 +8,282 @@
+  DateTimeRange datesOnly(covariant DateTimeRange range);
+
+  /// Returns true if the two [DateTime] objects have the same day, month, and
+  /// year, or are both null.InheritedWidget
+  bool isSameDay(covariant DateTime? dateA, covariant DateTime? dateB);
+
+  /// Returns true if the two [DateTime] objects have the same month and
+  /// year, or are both null.
+  bool isSameMonth(covariant DateTime? dateA, covariant DateTime? dateB);
+
+  /// Determines the number of months between two [DateTime] objects.
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// DateTime date1 = DateTime(2019, 6, 15);
+  /// DateTime date2 = DateTime(2020, 1, 15);
+  /// int delta = DateUtils.monthDelta(date1, date2);
+  /// ```
+  ///
+  /// The value for `delta` would be `7`.
+  int monthDelta(covariant DateTime startDate, covariant DateTime endDate);
+
+  /// Returns a [DateTime] that is [monthDate] with the added number
+  /// of months and the day set to 1 and time set to midnight.
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// DateTime date = DateTime(2019, 1, 15);
+  /// DateTime futureDate = DateUtils.addMonthsToMonthDate(date, 3);
+  /// ```
+  ///
+  /// `date` would be January 15, 2019.
+  /// `futureDate` would be April 1, 2019 since it adds 3 months.
+  DateTime addMonthsToMonthDate(covariant DateTime monthDate, int monthsToAdd);
+
+  /// Returns a [DateTime] with the added number of days and time set to
+  /// midnight.
+  DateTime addDaysToDate(covariant DateTime date, int days);
+
+  /// Computes the offset from the first day of the week that the first day of
+  /// the [month] falls on.
+  ///
+  /// For example, September 1, 2017 falls on a Friday, which in the calendar
+  /// localized for United States English appears as:
+  ///
+  ///     S M T W T F S
+  ///     _ _ _ _ _ 1 2
+  ///
+  /// The offset for the first day of the months is the number of leading blanks
+  /// in the calendar, i.e. 5.
+  ///
+  /// The same date localized for the Russian calendar has a different offset,
+  /// because the first day of week is Monday rather than Sunday:
+  ///
+  ///     M T W T F S S
+  ///     _ _ _ _ 1 2 3
+  ///
+  /// So the offset is 4, rather than 5.
+  ///
+  /// This code consolidates the following:
+  ///
+  /// - [DateTime.weekday] provides a 1-based index into days of week, with 1
+  ///   falling on Monday.
+  /// - [MaterialLocalizations.firstDayOfWeekIndex] provides a 0-based index
+  ///   into the [MaterialLocalizations.narrowWeekdays] list.
+  /// - [MaterialLocalizations.narrowWeekdays] list provides localized names of
+  ///   days of week, always starting with Sunday and ending with Saturday.
+  int firstDayOffset(int year, int month, MaterialLocalizations localizations);
+
+  /// Returns the number of days in a month, according to the proleptic
+  /// Gregorian calendar.
+  ///
+  /// This applies the leap year logic introduced by the Gregorian reforms of
```

Comment:
For the abstract class probably shouldn't refer to Gregorian logic.

Reply:
Removed ref to Gregorian logic